### PR TITLE
Update validation.py

### PR DESCRIPTION
**Pull request by akarloff**
## What changes are proposed in this pull request?
 
Add validation of string length for experiment name as defined in dbmodels/models.py.
 
Related issue: #1469 

## How is this patch tested?
 
Added unit tests.
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
Validate maximum length of experiment name is 256 characters or less. Throw error code INVALID_PARAMETER_VALUE if validation fails.
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [x] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/validation.py [hunk 1]
```diff
@@ -11,28 +11,30 @@
 from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
 from mlflow.store.dbmodels.db_types import DATABASE_ENGINES
 
-_VALID_PARAM_AND_METRIC_NAMES = re.compile(r"^[/\w.\- ]*$")
+_VALID_PARAM_AND_METRIC_NAMES = re.compile(r"^[/\w.\- ]{1,250}$")
```
[the remark this answers is not in the document]
Comment (akarloff) [reply mlflow/utils/validation.py]: Need to revert this change: https://github.com/mlflow/mlflow/issues/985

File: mlflow/utils/validation.py [hunk 2]
```diff
@@ -189,7 +191,11 @@ def _validate_experiment_name(experiment_name):
         raise MlflowException("Invalid experiment name: '%s'" % experiment_name,
                               error_code=INVALID_PARAMETER_VALUE)
     if not isinstance(experiment_name, str):
-        raise MlflowException("Invalid experiment name: %s. Expects a string." % experiment_name,
+        raise MlflowException("Invalid experiment name: '%s'. Expects a string." % experiment_name,
+                              error_code=INVALID_PARAMETER_VALUE)
+    if len(experiment_name) > MAX_EXPERIMENT_NAME_LENGTH:
+        raise MlflowException("Invalid experiment name: '%s'. Length must be %d characters or less"
+                              " in length." % (experiment_name, MAX_EXPERIMENT_NAME_LENGTH),
```
[the remark this answers is not in the document]
Comment (akarloff) [reply mlflow/utils/validation.py]: Calling _validate_length_limit() instead for consistent length errors.

File: mlflow/utils/validation.py [hunk 3]
```diff
@@ -18,21 +18,22 @@
...
 MAX_ENTITIES_PER_BATCH = 1000
 MAX_BATCH_LOG_REQUEST_SIZE = int(1e6)
 MAX_PARAM_VAL_LENGTH = 250
 MAX_TAG_VAL_LENGTH = 250
 MAX_ENTITY_KEY_LENGTH = 250
+MAX_EXPERIMENT_NAME_LENGTH = 256
 
 _UNSUPPORTED_DB_TYPE_MSG = "Supported database engines are {%s}" % ', '.join(DATABASE_ENGINES)
 
+_BAD_CHARACTERS_MESSAGE = (
```
Comment (eedeleon) [inline mlflow/utils/validation.py]: Is this only applied to names? Or also other strings/ids?

Comment (akarloff) [reply mlflow/utils/validation.py]: Thanks. Should just revert this.

**Review by tomasatdatabricks — COMMENTED** [review]
Hey @akarloff , thanks for the contribution.

The code looks good but can you elaborate on what is the motivation for this change? Is this an optimization or is the server side validation not sufficient in your case? 

I am asking because currently we do not specify a limit on length of the experiment name in the rest api.  Technically, different backends can have different limits and we did not feel like we should enforce this on the client side. In fact, enforcing the length limit on the client side would technically be a breaking change, since at least FileStore can currently work with longer names. Therefore, I don't think we should put this into the client. 

Let me know what you think.

**Comment by tomasatdatabricks** [comment]
I am closing this since there was no activity for a while.

**Pull request closed without merging**